force-app/main/default/classes/TaskService.cls

```diff
@@ -4,4 +4,46 @@ public with sharing class TaskService
     {
 
     }
+
+    /**************************************************************
+    *   Creates a task associated with the opportunity 
+    *   to follow up in 10 days time 
+    *   Called in the context of an after insert trigger
+    ***************************************************************/
+
+    public static void addFollowUpTask(Map<Id,Opportunity> oppMap)
+    {
+        //TODO: if(!Triggers.Stop) 
+
+        List<Task> newFollowUpTasks = new List<Task>();
+
+        for (Opportunity o : oppMap.values()) 
+        {
+            //TODO: if(o.StageName = 'Prospecting') //Ask KAB???
```

Comment:
There were no requirements around the opportunity stage, just that it should be when opportunities are created, so you are fine.